redhat-appstudio/service-provider-integration-operator
# initial APIResourceSchema and APIExport for KCP

### What does this PR do?
 - add `APIResourceSchema` for all our CRDs
 - add `APIExport` linking all `APIResourceSchema`s
 - add script to generate new version of `APIResourceSchema` on CRD change
 - make script runnable by `make manifests-kcp`, also added as dependency of `make manifests`

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-91

### How to test this PR?

script requires `kubectl-kcp` to run properly. I've tried to write simple instructions how to install it.

 - run `make manifests-kcp`, you should see something like `no changes in KCP API`
 - do some change in one of CRDs, change some field name for example
 - run `make manifests-kcp` again. you should see updated `apiresourceschema_spi.yaml` file with your changes and current datetime as a name prefix. `apiexport_spi.yaml` must contain updated schemas with latest datetime.
